### ISSUE TITLE
[explicit-module-boundary-types] considers "protected" members as public boundaries

### ISSUE THREAD

**Issue by bladeSk** [typescript-eslint/typescript-eslint#2430]
The rule considers `protected` class methods as `public`, which contradicts the description in the plugin's docs:

> Require explicit return and argument types on exported functions' and classes' public class methods

It's probably a matter of debate whether `protected` functions should be considered "exported" and I think the best solution would be to have this configurable.

This rule is generally pretty annoying, as it requires putting `: void` next to methods that don't return anything, but that's perhaps for another issue.

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": ["error"]
  }
}
```

```TS
export default class Test {
  testMethod() { // this produces an error, as expected
  }

  protected testMethod2() { // this produces an error, **but it shouldn't**
  }

  private testMethod3() { // this does not produce an error, as expected
  }
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.1` |
| `@typescript-eslint/parser`        | `3.9.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `v12.18.2` |

**Comment by bradzacher** [comment]
By definition - they are public.
They're not public in the sense that someone that constructs an instance can see them, but they're public in the sense that anyone that extends your class can see them.

If you are extending this class outside of this module, then they are classed as part of the public interface of the class.

I don't think allowing for configuration here is correct.

**Comment by tadhgmister** [comment]
I agree that protected members should be explicit since they are visible outside the module (even if it's only inside subclasses) but I do see where the confusion comes from, maybe just an update to the docs to say "public and protected members"

**Comment by bradzacher** [comment]
Happy to accept a PR to clarify that in the docs!
Closing this as we won't be changing the underlying rule implementation.
